### PR TITLE
react: add gateway monitor page

### PR DESCRIPTION
**Pull request by arthurescriou**
add gateway monitor page on the administration menu of a jhipster microservice gateway

add #6044

- Please make sure the below checklist is followed for Pull Requests.

- [ ] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
- [ ] Tests are added where necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

File: generators/client/templates/react/src/main/webapp/app/modules/administration/gateway/gateway.tsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+// import GatewayModal from './gateway-modal';
+
+export interface IGatewayPageProps {
+  getRoutes: ICrudGetAction;
+  routes: any[];
+  account: any;
+  isFetching?: boolean;
+}
+
+export class GatewayPage extends React.Component<IGatewayPageProps, any> {
```
Comment (deepu105) [inline generators/client/templates/react/src/main/webapp/app/modules/administration/gateway/gateway.tsx]: no need of second generic param for stateless componenet `React.Component<IGatewayPageProps>` is sufficient

File: generators/client/templates/react/src/main/webapp/app/modules/administration/gateway/gateway.tsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { Translate, ICrudGetAction } from 'react-jhipster';
+import { FaRefresh } from 'react-icons/lib/fa';
+
+import { getRoutes } from '../../../reducers/gateway';
+// import GatewayModal from './gateway-modal';
```
Comment (deepu105) [inline generators/client/templates/react/src/main/webapp/app/modules/administration/gateway/gateway.tsx]: does gateway has modal?

File: generators/client/templates/react/src/main/webapp/app/modules/administration/gateway/gateway.tsx [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+export interface IGatewayPageProps {
+  getRoutes: ICrudGetAction;
+  routes: any[];
+  account: any;
+  isFetching?: boolean;
+}
+
+export class GatewayPage extends React.Component<IGatewayPageProps, any> {
+
+  constructor(props) {
```
Comment (deepu105) [inline generators/client/templates/react/src/main/webapp/app/modules/administration/gateway/gateway.tsx]: not required for stateless component

File: generators/client/templates/react/src/main/webapp/app/modules/administration/gateway/gateway.tsx [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+  }
+
+  metadata = instance => {
+    const spans = [];
+    Object.keys(instance).map((key, index) => {
+      spans.push((
+        <span key={key.toString() + 'value'}>
+          <span key={key.toString() + '-containerbadge'} className="badge badge-default font-weight-normal">
+          <span key={key.toString() + '-badge'} className="badge badge-pill badge-info font-weight-normal">{key}</span>
+        {instance[key]}
```
Comment (deepu105) [inline generators/client/templates/react/src/main/webapp/app/modules/administration/gateway/gateway.tsx]: indent wrong

File: generators/client/templates/react/src/main/webapp/app/modules/administration/gateway/gateway.tsx [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+        return <div className="badge badge-danger">{info.status}</div>;
+      }
+    } else {
+      return <div className="badge badge-warning">?</div>;
+    }
+  }
+
+  instanceInfo = route => {
+    if (route) {
+    return (
```
Comment (deepu105) [inline generators/client/templates/react/src/main/webapp/app/modules/administration/gateway/gateway.tsx]: indent wrong

File: generators/client/templates/react/src/main/webapp/app/modules/administration/gateway/gateway.tsx [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+      <div className="table-responsive">
+      <table className="table table-striped">
```
Comment (deepu105) [inline generators/client/templates/react/src/main/webapp/app/modules/administration/gateway/gateway.tsx]: indent wrong

File: generators/client/templates/react/src/main/webapp/app/modules/administration/gateway/gateway.tsx [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+          <p>
+          <button type="button" onClick={this.getRoutes} className={isFetching ? 'btn btn-danger' : 'btn btn-primary'} disabled={isFetching}>
+            <FaRefresh />&nbsp;
+            <Translate component="span" contentKey="health.refresh.button" />
+          </button>
+          </p>
+          <div className="table-responsive">
+              <table className="table table-striped">
+              <thead>
+              <tr key="header">
```
Comment (deepu105) [inline generators/client/templates/react/src/main/webapp/app/modules/administration/gateway/gateway.tsx]: indent

File: generators/client/templates/react/src/main/webapp/app/reducers/index.ts [hunk 8]
```diff
@@ -8,6 +8,9 @@ import layout from './layout';
 import authentication from './authentication';
 import administration from './administration';
 import userManagement from './user-management';
+<%_ if (applicationType === 'gateway') { _%>
+import gateway from './gateway';
+  <%_ } _%>
```
Comment (deepu105) [inline generators/client/templates/react/src/main/webapp/app/reducers/index.ts]: indent

File: generators/client/templates/react/src/main/webapp/app/reducers/index.ts [hunk 9]
```diff
@@ -19,5 +22,8 @@ export default combineReducers({
   administration,
   userManagement,
   /* jhipster-needle-add-reducer-combine - JHipster will add reducer here */
+  <%_ if (applicationType === 'gateway') { _%>
+  gateway,
+    <%_ } _%>
```
Comment (deepu105) [inline generators/client/templates/react/src/main/webapp/app/reducers/index.ts]: indent

File: generators/client/templates/react/src/main/webapp/app/shared/layout/header/header.tsx [hunk 10]
```diff
@@ -7,10 +7,11 @@ import {
   UncontrolledNavDropdown, DropdownToggle, DropdownMenu, DropdownItem
 } from 'reactstrap';
 import {
-  FaHome, FaThList, FaUserPlus, FaUser, FaFlag, FaHeart,
-  FaList, FaTasks, FaDashboard, FaBook, FaWrench, FaSignIn, FaSignOut,
-  FaClockO, FaHddO, FaAsterisk
-} from 'react-icons/lib/fa';
+    FaHome, FaThList, FaUserPlus, FaUser, FaFlag, FaHeart,
```
Comment (deepu105) [inline generators/client/templates/react/src/main/webapp/app/shared/layout/header/header.tsx]: indent wrong

File: generators/client/templates/react/src/main/webapp/app/modules/administration/gateway/gateway.tsx [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+          <table className="table table-striped">
+            <tbody>
+              {route.serviceInstances.map((instance, i) =>
+                <tr key={instance.instanceInfo + '-info'}>
+                  <td><a href={instance.uri} target="_blank">{instance.uri}</a></td>
+                  <td>{this.badgeInfo(instance.instanceInfo)}</td>
+                  <td>{this.metadata(instance.metadata)}</td>
+                </tr>
+              )}
+        </tbody>
```
Comment (deepu105) [inline generators/client/templates/react/src/main/webapp/app/modules/administration/gateway/gateway.tsx]: indentation is wrong on these 2 lines

File: generators/client/templates/react/src/main/webapp/app/modules/administration/gateway/gateway.tsx [hunk 12]
```diff
@@ -0,0 +1,117 @@
...
+              )}
+              </tbody>
+            </table>
+          </div>
+        </div>
+    );
+  }
+}
+
+const mapStateToProps = storeState => ({
```
Comment (deepu105) [inline generators/client/templates/react/src/main/webapp/app/modules/administration/gateway/gateway.tsx]: deconstruct the object `({ gateway, authentication })`

File: generators/client/templates/react/src/main/webapp/app/reducers/gateway.ts [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+        routes: action.payload.data
+      };
+    default:
+      return state;
+  }
+};
+
+const apiUrl = '/api/gateway/routes';
+// Actions
+export const getRoutes: ICrudGetAction = () => ({
```
Comment (deepu105) [inline generators/client/templates/react/src/main/webapp/app/reducers/gateway.ts]: check adminitartion.ts, if it has method for this remove it there

File: generators/client/templates/react/src/main/webapp/app/shared/layout/header/header.tsx [hunk 14]
```diff
@@ -9,8 +9,10 @@ import {
 import {
   FaHome, FaThList, FaUserPlus, FaUser, FaFlag, FaHeart,
   FaList, FaTasks, FaDashboard, FaBook, FaWrench, FaSignIn, FaSignOut,
-  FaClockO, FaHddO, FaAsterisk
+  FaClockO, FaHddO, FaAsterisk<%_ if (applicationType === 'gateway') { _%>, FaRoad<%_ } _%>
```
Comment (deepu105) [inline generators/client/templates/react/src/main/webapp/app/shared/layout/header/header.tsx]: `<% %>` as <%_` is not for inline

File: generators/client/templates/react/src/main/webapp/app/modules/administration/gateway/gateway.tsx [hunk 15]
```diff
@@ -0,0 +1,116 @@
...
+}
+
+export class GatewayPage extends React.Component<IGatewayPageProps> {
+
+  componentDidMount() {
+    this.props.gatewayRoutes();
+  }
+
+  metadata = instance => {
+    const spans = [];
```
Comment (deepu105) [inline generators/client/templates/react/src/main/webapp/app/modules/administration/gateway/gateway.tsx]: shoudld be simply
```
metadata = instance => Object.keys(instance).map((key, index) => (
  <span key={key.toString() + 'value'}>...</span>
));
```

File: generators/client/templates/react/src/main/webapp/app/modules/administration/gateway/gateway.tsx [hunk 16]
```diff
@@ -0,0 +1,116 @@
...
+                <tr key={instance.instanceInfo + '-info'}>
+                  <td><a href={instance.uri} target="_blank">{instance.uri}</a></td>
+                  <td>{this.badgeInfo(instance.instanceInfo)}</td>
+                  <td>{this.metadata(instance.metadata)}</td>
+                </tr>
+              )}
+            </tbody>
+          </table>
+        </div>);
+    }
```
Comment (deepu105) [inline generators/client/templates/react/src/main/webapp/app/modules/administration/gateway/gateway.tsx]: else return null

**Comment by sendilkumarn** [comment]
I have noticied the first one but these are few things that we have to anyhow change. 

For the second, 🤔  There is no error in the frontend, when running.

**Pull request merged**